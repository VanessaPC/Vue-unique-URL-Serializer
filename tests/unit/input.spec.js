import { render, screen, fireEvent } from "@testing-library/vue";
import "@testing-library/jest-dom/extend-expect";

import UserInputURL from "@/components/input/index.vue";

describe("UserInputURL and submit button behaviour", () => {
  it("should render UserInputURL", () => {
    const { container } = render(UserInputURL);
    expect(container).toMatchSnapshot();
  });

  it("if input field is empty it should have the input field and not the button ", () => {
    render(UserInputURL);

    const input = screen.getByPlaceholderText("Enter your url");

    fireEvent.change(input, {
      target: { value: "thisnewURL.com" },
    });

    const button = screen.queryByText("Submit URL");

    expect(button).not.toBeVisible();
  });

  it("it should have the input field, and fill it, and then the button should be visible ", async () => {
    render(UserInputURL);

    const input = screen.getByPlaceholderText("Enter your url");

    await fireEvent.update(input, "thisnewURL.com");

    const button = screen.queryByText("Submit URL");

    expect(button).toBeVisible();
  });
});
