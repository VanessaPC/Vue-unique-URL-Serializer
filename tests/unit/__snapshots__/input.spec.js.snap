// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInputURL and submit button behaviour should render UserInputURL 1`] = `
<div>
  <div>
    <div
      class="kYYMJh"
    >
      <input
        class="hwMher"
        placeholder="Enter your url"
      />
       
      <button
        class="gGPVhI"
        style="display: none;"
        type="submit"
      >
        Submit URL
      </button>
    </div>
     
    <div
      class="GFraz"
    >
      <span
        style="display: none;"
      >
        This url already exists: 
      </span>
       
      <span
        style="display: none;"
      >
        https://pbid.io/
      </span>
    </div>
  </div>
</div>
`;
